prefect_databricks/flows.py

```diff
@@ -306,18 +340,76 @@ async def jobs_runs_submit_and_wait_for_completion_flow(notebook_path, **base_pa
                 f"encountered an internal error: {jobs_runs_state_message}.",
             )
         else:
-            logger.info(
-                "Databricks Jobs Runs Submit (%s ID %s) has %s state. "
-                "Waiting for %i seconds.",
-                run_name,
-                multi_task_jobs_runs_id,
-                jobs_runs_life_cycle_state.lower(),
-                poll_frequency_seconds,
-            )
+            logger.info("Waiting for %s seconds.", poll_frequency_seconds)
             await asyncio.sleep(poll_frequency_seconds)
             seconds_waited_for_run_completion += poll_frequency_seconds
 
     raise DatabricksJobRunTimedOut(
         f"Max wait time of {max_wait_seconds} seconds exceeded while waiting "
         f"for job run ({run_name} ID {multi_task_jobs_runs_id})"
     )
+
+
+def __log_state(
+    array: Dict[str, Any],
+    run_id: str,
+    state: Dict[str, Any],
+    run_page_url: str,
+    logger: Logger,
+    is_task: bool = True,
+):
+    """
+    Stores the states of a job or task to its collection and logs the output
+    if it changes
+
+    Args:
+        array: The array of the job or task collection.
+        run_id: Id of the run in Databricks for the job or task.
+        state: State object of the run in Databricks for the job or task.
+        run_page_url: URL of the run in Databricks for the job or task in
+            the Databricks UI.
+        logger: Logger instance to log with.
+        is_task: Bool indicating if is job or task
+
+    """
+
+    string_run_id = str(run_id)
+
+    if string_run_id not in array:
+        array[string_run_id] = {}
+
+    run_type = "Task" if is_task else "Job"
+
+    new_item = {}
+    new_item["run_page_url"] = run_page_url
+    new_item["run_id"] = run_id
+    new_item["state"] = state
+
+    life_cycle_state = state.get("life_cycle_state", "")
+    state_message = state.get("state_message", "")
+    result_state = state.get("result_state", "")
+
+    if "state" in array[string_run_id]:
+        existing_state = array[string_run_id]["state"]
+        existing_life_cycle_state = existing_state.get("life_cycle_state", "")
+        existing_state_message = existing_state.get("state_message", "")
+        existing_result_state = existing_state.get("result_state", "")
+
+        if (
+            life_cycle_state == existing_life_cycle_state
+            and state_message == existing_state_message
+            and result_state == existing_result_state
+        ):
+            return
+
+    logger.info(
+        "%s Run '%s' transitioned state. '%s', '%s' with message '%s':  %s",
+        run_type,
+        run_id,
+        life_cycle_state,
+        result_state,
+        state_message,
+        run_page_url,
+    )
+
+    array[string_run_id] = new_item
```

Comment:
I'd advise against changing the contents of a passed in dictionary. I'd recommend either returning a new dictionary with the desired changes included, or declaring a dictionary for use in this module and removing the need for the caller to pass in a dictionary for tracking state between calls.